packages/ember-data/lib/system/model/model.js

```diff
@@ -781,7 +782,11 @@ var Model = Ember.Object.extend(Ember.Evented, {
 
   hasMany: function() {
     Ember.assert("The `hasMany` method is not available on DS.Model, a DS.Snapshot was probably expected. Are you passing a DS.Model instead of a DS.Snapshot to your serializer?", false);
-  }
+  },
+
+  setId: Ember.observer('id', function () {
```

Comment:
in the tests bellow, we only update the ID via:

https://github.com/emberjs/data/blob/0f06c86b881ffb22e3e3fd8b2ef61726b9c17b22/packages/ember-data/lib/system/store.js#L1394-L1403

which already calls [internalModel.setId](https://github.com/emberjs/data/blob/0f06c86b881ffb22e3e3fd8b2ef61726b9c17b22/packages/ember-data/lib/system/store.js#L1402)